```
# git clone https://github.com/haosdent/vllm -b fix-35057
# git cherry-pick b94630eab94b56da9c659c5f0de2e7455687b1c3
# export VLLM_PRECOMPILED_WHEEL_VARIANT="cu129"
# export VLLM_USE_PRECOMPILED=1
# export VLLM_PRECOMPILED_WHEEL_COMMIT="7291d1b288558d48508e1a17c37b0aa170332264"
# uv pip install --system --editable .
# python3 -m vllm.entrypoints.openai.api_server --port 5000 --host 0.0.0.0 --download-dir /workspace/.cache/huggingface/hub --model Qwen/Qwen3.5-397B-A17B-FP8 --tensor-parallel-size 4 --decode-context-parallel-size 2 --trust-remote-code --enable-chunked-prefill --enable-prefix-caching --max-num-seqs 128 --gpu-memory-utilization 0.95 --hf-overrides '{"text_config": {"max_position_embeddings": 1010000, "rope_parameters": {"mrope_interleaved": true, "mrope_section": [11, 11, 10], "rope_type": "yarn", "rope_theta": 10000000, "partial_rotary_factor": 0.25, "factor": 4.0, "original_max_position_embeddings": 262144}}}' --max-model-len 1010000 --enable-auto-tool-choice --tool-call-parser qwen3_coder --reasoning-parser qwen3 --mm-processor-cache-gb 0 --mm-encoder-tp-mode data -cc.pass_config.fuse_allreduce_rms=False
```

Failure logs:

<details>
<summary>Open</summary>

```
(EngineCore_DP0 pid=11331) ERROR 02-23 10:27:11 [core.py:1029] EngineCore failed to start.
(EngineCore_DP0 pid=11331) ERROR 02-23 10:27:11 [core.py:1029] Traceback (most recent call last):
(EngineCore_DP0 pid=11331) ERROR 02-23 10:27:11 [core.py:1029]   File "/workspace/vllm/vllm/v1/engine/core.py", line 1019, in run_engine_core
(EngineCore_DP0 pid=11331) ERROR 02-23 10:27:11 [core.py:1029]     engine_core = EngineCoreProc(*args, engine_index=dp_rank, **kwargs)
(EngineCore_DP0 pid=11331) ERROR 02-23 10:27:11 [core.py:1029]                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
(EngineCore_DP0 pid=11331) ERROR 02-23 10:27:11 [core.py:1029]   File "/workspace/vllm/vllm/tracing/otel.py", line 178, in sync_wrapper
(EngineCore_DP0 pid=11331) ERROR 02-23 10:27:11 [core.py:1029]     return func(*args, **kwargs)
(EngineCore_DP0 pid=11331) ERROR 02-23 10:27:11 [core.py:1029]            ^^^^^^^^^^^^^^^^^^^^^
(EngineCore_DP0 pid=11331) ERROR 02-23 10:27:11 [core.py:1029]   File "/workspace/vllm/vllm/v1/engine/core.py", line 763, in __init__
(EngineCore_DP0 pid=11331) ERROR 02-23 10:27:11 [core.py:1029]     super().__init__(
(EngineCore_DP0 pid=11331) ERROR 02-23 10:27:11 [core.py:1029]   File "/workspace/vllm/vllm/v1/engine/core.py", line 140, in __init__
(EngineCore_DP0 pid=11331) ERROR 02-23 10:27:11 [core.py:1029]     self.scheduler: SchedulerInterface = Scheduler(
(EngineCore_DP0 pid=11331) ERROR 02-23 10:27:11 [core.py:1029]                                          ^^^^^^^^^^
(EngineCore_DP0 pid=11331) ERROR 02-23 10:27:11 [core.py:1029]   File "/workspace/vllm/vllm/v1/core/sched/async_scheduler.py", line 14, in __init__
(EngineCore_DP0 pid=11331) ERROR 02-23 10:27:11 [core.py:1029]     super().__init__(*args, **kwargs)
(EngineCore_DP0 pid=11331) ERROR 02-23 10:27:11 [core.py:1029]   File "/workspace/vllm/vllm/v1/core/sched/scheduler.py", line 221, in __init__
(EngineCore_DP0 pid=11331) ERROR 02-23 10:27:11 [core.py:1029]     self.kv_cache_manager = KVCacheManager(
(EngineCore_DP0 pid=11331) ERROR 02-23 10:27:11 [core.py:1029]                             ^^^^^^^^^^^^^^^
(EngineCore_DP0 pid=11331) ERROR 02-23 10:27:11 [core.py:1029]   File "/workspace/vllm/vllm/v1/core/kv_cache_manager.py", line 119, in __init__
(EngineCore_DP0 pid=11331) ERROR 02-23 10:27:11 [core.py:1029]     self.coordinator = get_kv_cache_coordinator(
(EngineCore_DP0 pid=11331) ERROR 02-23 10:27:11 [core.py:1029]                        ^^^^^^^^^^^^^^^^^^^^^^^^^
(EngineCore_DP0 pid=11331) ERROR 02-23 10:27:11 [core.py:1029]   File "/workspace/vllm/vllm/v1/core/kv_cache_coordinator.py", line 581, in get_kv_cache_coordinator
(EngineCore_DP0 pid=11331) ERROR 02-23 10:27:11 [core.py:1029]     return HybridKVCacheCoordinator(
(EngineCore_DP0 pid=11331) ERROR 02-23 10:27:11 [core.py:1029]            ^^^^^^^^^^^^^^^^^^^^^^^^^
(EngineCore_DP0 pid=11331) ERROR 02-23 10:27:11 [core.py:1029]   File "/workspace/vllm/vllm/v1/core/kv_cache_coordinator.py", line 402, in __init__
(EngineCore_DP0 pid=11331) ERROR 02-23 10:27:11 [core.py:1029]     assert all(
(EngineCore_DP0 pid=11331) ERROR 02-23 10:27:11 [core.py:1029]            ^^^^
(EngineCore_DP0 pid=11331) ERROR 02-23 10:27:11 [core.py:1029] AssertionError: block_size must be divisible by hash_block_size
```

</details>

--- a/vllm/v1/attention/backends/flash_attn.py
+++ b/vllm/v1/attention/backends/flash_attn.py
@@ -847,6 +847,7 @@ class FlashAttentionImpl(AttentionImpl):
             q_descale=q_descale,
             k_descale=k_descale,
             v_descale=v_descale,
+            num_splits=attn_metadata.max_num_splits,
         )
         # FA returns LSE in shape [ H, B ] but cp_lse_ag_out_rs wants [ B, H ]
         context_attn_out_cor, context_lse_cor = cp_lse_ag_out_rs(
@@ -876,6 +877,7 @@ class FlashAttentionImpl(AttentionImpl):
             q_descale=q_descale,
             k_descale=k_descale,
             v_descale=v_descale,
+            num_splits=attn_metadata.max_num_splits,
         )
         assert context_attn_out_cor.shape == query_attn_out.shape
         assert context_lse_cor.shape == query_lse.shape

